### DISCUSSION TITLE
Filament 3 form modal with autofocus and ...again

### DISCUSSION THREAD

**Discussion by cyles** [filamentphp/filament#11399] [Help] [answered]
### Package

Actions

### Package Version

v3.2.31

### How can we help you?

I have a Filament 3 app with multiple tables and working relationships.  In one table, I have a custom action that pops up a modal with a TextInput.  Based on the contents of that field, it looks up a record and then sync's to a many-many pivot table between the table row where the action started and the result of the lookup.  This all works fine.

There are two features I still need that I haven't been able to work out:

1. The text field needs to autofocus when the modal appears.  I've tried ->autofocus(), but that doesn't really work.  It will rarely work, but not every time.
2. I need to implement the 'and ___ another' button (like from create or attach) as the default submit action for this form and not have the singular submit button.  I know this is possible because create and attach built-in actions do it, but I haven't been able to work out how.

I tried going down a custom page route (which does the autofocus fine since it's on a page load), but the examples out there seem to be for older Filament and don't seem to match up with the current version.  At any rate, I can't work out how to pass data back and forth to a custom page.  However, I would much rather do this as 'native' filament rather than a custom page.

The next step is to add another child page to 'register' a record if the textInput is not found.  This seems well documented, so I would like to stay in Filament to do this bit.

Thanks for any help!

**Comment by cyles** [comment]
Here's the action code from my table.
```
Action::make('checkin')
                    ->form([
                        Forms\Components\TextInput::make('cardID')
                            ->label('Card ID')
                            ->autofocus() // This doesn't work - because it's a modal?
                            ->required(),
                    ])
                    ->modalSubmitAction(false) // turns off the 'singlular' submit button
                    ->extraModalFooterActions(fn (Action $action): array => [
                        $action->makeModalSubmitAction('checkinAnother', arguments: ['another' => true]),
                    ])
                    ->action(function (array $data, array $arguments, Timeslot $record): void {
                        if (Student::where('cardID', $data)->exists()){
                            $record->students()->attach(Student::where('cardID', $data)->first()->id);
                            Notification::make()
                            ->title('CardID found')
                            ->send();
                        }
                        else{
                            Notification::make()
                            ->title('CardID NOT found')
                            ->send();
                        }

                        if ($arguments['another'] ?? false) {
                            // Reset the form and don't close the modal - guessing here, not in docs
                            $this->callAfter(); //ERROR - Using $this when not in object context
                            $this->record(null); //Same error - removing $this results in undefined method error
                            $this->form->fill();  // This doesn't work either - form undefined
                            $this->halt();
                            return;
                        }
                    })
```

**Reply by cyles** [reply]
The If ($arguments['another'] section was [taken from the docs](https://filamentphp.com/docs/3.x/actions/modals#adding-an-extra-modal-action-button-to-the-footer) : and the contents from [.../packages/tables/src/Actions/AttachAction.php] (https://github.com/filamentphp/filament/tree/e806b25b5dbd48490058556a6fcb9dbd70173e2c/packages/tables/src/Actions/AttachAction.php)
I wonder if I need this as the AttachAction supposedly does this for me?  However, when I take this if statement out of my table action, it doesn't 'repeat' the modal form to accept another input.

What do I need to put after "//Reset the form and don't close the modal" to get this to work?

**Reply by leandrocfe** [reply, marked as answer]
You should inject the action and the form

```php
->action(function (array $data, array $arguments, Timeslot $record, Tables\Actions\Action $action, Form $form): void {

    $form->fill();
    $action->halt();
    
})
```

**Reply by cyles** [reply]
Awesome, thank you very much!

**Reply by hazwan91** [reply]
@cyles have you solved on the autofocus issue?

**Reply by albzour** [reply]
> @cyles have you solved on the autofocus issue?

This should work
```
Action::make('action')
    ->extraModalWindowAttributes(['x-trap.noscroll' => 'isShown'])
```

**Reply by dardarwail** [reply]
Thanks @albzour (y)
